### PR TITLE
W-11209778: Validation to prevent flow-refs inside Mule Operations

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

**Comment by mbuchwald** [comment]
--validate

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ */
+package org.mule.runtime.config.internal.validation;
+
+import org.junit.Test;
+import org.mule.runtime.ast.api.validation.Validation;
+import org.mule.runtime.ast.api.validation.ValidationResultItem;
+
+import java.util.Optional;
+
+import static java.util.Optional.empty;
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java:15]: static impots go first

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.validation;
+
+import org.junit.Test;
+import org.mule.runtime.ast.api.validation.Validation;
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java:10]: mule imports go first

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+
+public class OperationDoesNotHaveFlowRefTestCase extends AbstractCoreValidationTestCase {
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java:21]: allure

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.validation;
+
+import static java.util.Optional.empty;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+import static org.mule.test.allure.AllureConstants.MuleDsl.DslValidationStory.DSL_VALIDATION_STORY;
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java:14]: mule imports go firsr

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import static org.mule.test.allure.AllureConstants.MuleDsl.MULE_DSL;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.mule.runtime.ast.api.validation.Validation;
+import org.mule.runtime.ast.api.validation.ValidationResultItem;
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java:21]: import order: mule, java, the rest

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.Optional;
+import org.junit.Test;
+
+@Feature(MULE_DSL)
+@Story(DSL_VALIDATION_STORY)
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java:26]: add the mule operations feature/story as well

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRef.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+public class OperationDoesNotHaveFlowRef implements Validation {
+
+  private static final String FLOW_REF_ELEMENT = "flow-ref";
+  private static final ComponentIdentifier FLOW_REF_IDENTIFIER =
+      builder().namespace(CORE_PREFIX).name(FLOW_REF_ELEMENT).build();
+  private static final ComponentIdentifier OPERATION_IDENTIFIER = builder().namespace("operation").name("def").build();
+
+  @Override
+  public String getName() {
+    return "Operation don't have flow-ref";
```
Comment (sofiamorseletto) [inline runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRef.java:38]: doesn't*

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+import org.mule.runtime.ast.api.validation.Validation;
+import org.mule.runtime.ast.api.validation.ValidationResultItem;
+import java.util.Optional;
+import org.junit.Test;
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+
+@Feature(MULE_DSL)
+@Feature(REUSE)
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java:28]: group with `@Features`

File: runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+@Story(DSL_VALIDATION_STORY)
+@Story(OPERATIONS)
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveFlowRefTestCase.java:30]: group with `@Stories`

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**